src/CanvasPattern.cc

```diff
@@ -4,122 +4,115 @@
+    return;
   }
 
+  Napi::Object obj = info[0].As<Napi::Object>();
+  InstanceData* data = env.GetInstanceData<InstanceData>();
   cairo_surface_t *surface;
 
-  Local<Object> obj = Nan::To<Object>(info[0]).ToLocalChecked();
-
   // Image
-  if (Nan::New(Image::constructor)->HasInstance(obj)) {
-    Image *img = Nan::ObjectWrap::Unwrap<Image>(obj);
+  if (obj.InstanceOf(data->ImageCtor.Value()).UnwrapOr(false)) {
+    Image *img = Image::Unwrap(obj);
     if (!img->isComplete()) {
-      return Nan::ThrowError("Image given has not completed loading");
+      Napi::Error::New(env, "Image given has not completed loading").ThrowAsJavaScriptException();
+      return;
     }
     surface = img->surface();
 
   // Canvas
-  } else if (Nan::New(Canvas::constructor)->HasInstance(obj)) {
-    Canvas *canvas = Nan::ObjectWrap::Unwrap<Canvas>(obj);
+  } else if (obj.InstanceOf(data->CanvasCtor.Value()).UnwrapOr(false)) {
+    Canvas *canvas = Canvas::Unwrap(obj);
     surface = canvas->surface();
   // Invalid
   } else {
-    return Nan::ThrowTypeError("Image or Canvas expected");
+    if (!env.IsExceptionPending()) {
+      Napi::TypeError::New(env, "Image or Canvas expected").ThrowAsJavaScriptException();
+    }
+    return;
   }
-  repeat_type_t repeat = REPEAT;
-  if (0 == strcmp("no-repeat", *Nan::Utf8String(info[1]))) {
-    repeat = NO_REPEAT;
-  } else if (0 == strcmp("repeat-x", *Nan::Utf8String(info[1]))) {
-    repeat = REPEAT_X;
-  } else if (0 == strcmp("repeat-y", *Nan::Utf8String(info[1]))) {
-    repeat = REPEAT_Y;
+  _pattern = cairo_pattern_create_for_surface(surface);
+
+  if (info[1].IsString()) {
+    if ("no-repeat" == info[1].As<Napi::String>().Utf8Value()) {
+      _repeat = NO_REPEAT;
+    } else if ("repeat-x" == info[1].As<Napi::String>().Utf8Value()) {
+      _repeat = REPEAT_X;
+    } else if ("repeat-y" == info[1].As<Napi::String>().Utf8Value()) {
+      _repeat = REPEAT_Y;
+    }
   }
-  Pattern *pattern = new Pattern(surface, repeat);
-  pattern->Wrap(info.This());
-  info.GetReturnValue().Set(info.This());
+
+  cairo_pattern_set_user_data(_pattern, pattern_repeat_key, &_repeat, NULL);
 }
 
 /*
  * Set the pattern-space to user-space transform.
  */
-NAN_METHOD(Pattern::SetTransform) {
-  Pattern *pattern = Nan::ObjectWrap::Unwrap<Pattern>(info.This());
-  Local<Context> ctx = Nan::GetCurrentContext();
-  Local<Object> mat = Nan::To<Object>(info[0]).ToLocalChecked();
+void
+Pattern::setTransform(const Napi::CallbackInfo& info) {
+  if (!info[0].IsObject()) {
+    Napi::TypeError::New(env, "Expected DOMMatrix").ThrowAsJavaScriptException();
+    return;
+  }
+
+  Napi::Object mat = info[0].As<Napi::Object>();
 
 #if NODE_MAJOR_VERSION >= 8
-  if (!mat->InstanceOf(ctx, _DOMMatrix.Get(Isolate::GetCurrent())).ToChecked()) {
-    return Nan::ThrowTypeError("Expected DOMMatrix");
+  InstanceData* data = env.GetInstanceData<InstanceData>();
+  if (!mat.InstnaceOf(data->DOMMatrix.Instance)) {
```

Comment:
```suggestion
  if (!mat.InstanceOf(data->DOMMatrixCtor.Value()).UnwrapOr(false)) {
```

Reply:
Wow, this typo indicates that the macro above (`NODE_MAJOR_VERSION`) was not working